examples/plot_pvlaues.py

```diff
@@ -0,0 +1,186 @@
+
+###############################################################################
+# --- run linear regression analysis using scikit-learn ---
+# data to be analysed
+data = limo_epochs['2'].get_data(picks_eeg)
+
+# number of epochs in data set
+n_epochs = data.shape[0]
+
+# number of channels and number of time points in each epoch
+# we'll use this information later to bring the results of the
+# the linear regression algorithm into an eeg-like format
+# (i.e., channels x times points)
+n_channels = picks_eeg.shape[0]
+n_times = len(limo_epochs['2'].times)
+
+# number of trials and number of predictors
+n_trials, n_predictors = design.shape
+# degrees of freedom
+dfs = float(n_trials - n_predictors)
+
+# vectorize (channel) data for linear regression
+Y = Vectorizer().fit_transform(data)
+
+# set up model and fit linear model
+linear_model = LinearRegression(fit_intercept=False)
+linear_model.fit(design, Y)
+
+###############################################################################
+# extract the coefficients for linear model estimator
+betas = get_coef(linear_model, 'coef_')
+
+# compute predictions
+predictions = linear_model.predict(design)
+
+# compute sum of squared residuals and mean squared error
+residuals = (Y - predictions)
+# sum of squared residuals
+ssr = np.sum(residuals ** 2, axis=0)
+# mean squared error
+sqrt_mse = np.sqrt(ssr / dfs)
+
+# raw error terms for each predictor in the design matrix:
+# here, we take the inverse of the design matrix's projections
+# (i.e., A^T*A)^-1 and extract the square root of the diagonal values.
+error_terms = np.sqrt(np.diag(np.linalg.pinv(np.dot(design.T, design))))
+
+###############################################################################
+# define dictionaries for results
+lm_betas, stderrs, t_vals, p_vals, s_vals = (dict() for _ in range(5))
+
+# define point asymptotic to zero to use as zero
+tiny = np.finfo(np.float64).tiny
+
+# loop through predictors to extract parameters
+for ind, predictor in enumerate(predictors):
+
+    # extract coefficients for predictor in question
+    beta = betas[:, ind]
+    # compute standard errors
+    stderr = sqrt_mse * error_terms[ind]
+
+    # compute t values
+    t_val = beta / stderr
+    # and p-values
+    p_val = 2 * stats.t.sf(np.abs(t_val), dfs)
+
+    # project results back to channels x time points space
+    beta = beta.reshape((n_channels, n_times))
+    stderr = stderr.reshape((n_channels, n_times))
+    t_val = t_val.reshape((n_channels, n_times))
+    # replace p-values == 0 with asymptotic value `tiny`
+    p_val = np.clip(p_val, tiny, 1.).reshape((n_channels, n_times))
+
+    # create evoked object for plotting
+    lm_betas[predictor] = EvokedArray(beta, epochs_info, tmin)
+    stderrs[predictor] = EvokedArray(stderr, epochs_info, tmin)
+    t_vals[predictor] = EvokedArray(t_val, epochs_info, tmin)
+    p_vals[predictor] = p_val
+    # transform p-values to Shannon information values (i.e., surprise values)
```

Comment:
Isn't this a bit under explained? Add some info maybe, a reference even.

Reply:
You are right. I added a little bit more explanation and a reference for further reading.